content/docs/specs/native_histograms.md

```diff
@@ -0,0 +1,2353 @@
+simply renders it with the same width as the regular exponential buckets (which
+in turn means that the _x_ axis is not strictly exponential around the zero
+point). (TODO: How to do the rendering for non-exponential schemas.)
+
+With a linear _x_ axis, the buckets generally have varying width. Therefore,
+the _y_ axis displays the bucket population divided by its width. The
+Prometheus UI does not render values on the _y_ axis as they would be hard to
+interpret for humans anyway. The population can still be inspected in the text
+representation.
+
+In the _Graph_ view, Prometheus displays a heatmap (TODO: not yet, see below),
+which could be seen as a series of histograms over time, rotated by 90 degrees
+and encoding the bucket population as a color rather than the height of a bar.
+The typical query to render a counter-like histogram as a heatmap would be a
+`rate` query. A heatmap is an extremely powerful representation that allows
+humans to easily spot characteristics of distributions as they change over
+time.
+
+TODO: Heatmaps are not implemented yet. Instead, the UI plots just the sum of
+observations as a conventional graph. See [tracking
+issue](https://github.com/prometheus/prometheus/issues/11268). The same issue
+also discusses how to deal with the rendering of range vectors in the _Table_
+view.
+
+## Template expansion
+
+Native histograms work in template expansion. They are rendered in a text
+representation inspired by the mathematical notation of open and closed
+intervals. (This is generated by the `FloatHistogram.String` method in Go.) As
+native histograms can have a lot of buckets and bucket boundaries tend to have
+boundaries with a lot of decimal places, the representation isn't necessarily
+very readable. Use native histograms in template expansion judiciously.
+
+Example for the text representation of a float histogram:
+
+```
+{count:3493.3, sum:2.349209324e+06, [-22.62741699796952,-16):1000, [-16,-11.31370849898476):123400, [-4,-2.82842712474619):3, [-2.82842712474619,-2):3.1, [-0.01,0.01]:5.5, (0.35355339059327373,0.5]:1, (1,1.414213562373095]:3.3, (1.414213562373095,2]:4.2, (2,2.82842712474619]:0.1}
+```
+
+## Remote write & read
+
+The [protobuf specs for remote write &
+read](https://github.com/prometheus/prometheus/blob/main/prompb) were extended
+for native histograms as an experimental feature. Receivers not capable of
+processing native histograms will simply ignore the newly added fields.
+Nevertheless, Prometheus has to be configured to send native histograms via
+remote write (by setting the `send_native_histograms` remote write config
+setting to true).
+
+In [remote write v2](https://prometheus.io/docs/specs/remote_write_spec_2_0/),
+native histograms are a stable feature.
+
+TODO: Is there more to say? I'm not a remote write & read specialist. Does
+remote read really work?
+
+TODO: What are the plans to auto-convert classic histograms to NHCBs in
```

Comment:
No plan at the moment because we cannot guarantee that the classic histogram came from scrape where we for sure know we have all the buckets present. This is true for both receiving and sending remote write. In fact one of the reason for having NHCB is to avoid scenarios where classic histograms are not consistent (missing/late due to independent series). Therefor currently the only plans are to have NHCB when we know we have all data : scrape - being implemented, receiving OTEL histogram - issue there, expose from instrumentation directly.